### DISCUSSION TITLE
Support both javaAgent and JavaSdk for distributed tracking in the same process and different process

### DISCUSSION THREAD

**Discussion by chenyi19851209** [apache/skywalking#8261] [Q&A] [answered]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/skywalking/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

When we promote tracing in a distributed system, we found that some java system users are willing to accept the javaAgent mode, and some java system users are more willing to accept the javaSDK embedding mode based on risk or security considerations. I know that skywalking provides trace toolkit. This kind of annotation is for the business system to embed, but it also needs to rely on the bytecode enhancement implantation of javaAgent. Does skywalking's javaAgent consider extracting a javaSDK for tracking like opentelemetry-SdK, allowing users to choose between javaAgent or javaSDK tracking in the same process or between different processes?

### Use case

When we promote tracing in a distributed system, we found that some java system users are willing to accept the javaAgent mode, and some java system users are more willing to accept the javaSDK embedding mode based on risk or security considerations.

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by wu-sheng** [comment, marked as answer]
Hi, you don't need SkyWalking's approval to write a SDK for Java or any other languages. You can see many SDKs are written by different people in the SkyAPM org. You definitely could write one for open source or privately.

**Reply by chenyi19851209** [reply]
We are currently trying to extract a java SDK based on skywalking javaAgent. When it is done, will the skywalking community consider accepting the pr? The premise of extracting the javaSDK is that the current apm-agent-core module will be significantly modified and refactored

**Reply by wu-sheng** [reply]
It depends how many parts you are going to change. If too many changes are applied, I would recommend you to open source another repository and release on your own. Currently, not enough PMC member(actually not a single one yet) is asking for this.
Unless you could find enough SkyWalking PMC members(>= 3) saying they could cover it, this should be a 3rd-party repository.

**Reply by wu-sheng** [reply]
This standard is as same as every new subproject creation in the SkyWalking, because you at least need 3 PMC members to vote +1 for official release.

**Reply by chenyi19851209** [reply]
OK. I understand, thanks
